### PR TITLE
WFCORE-803 WFCORE-912 register capabilities on resources

### PR DESCRIPTION
**Pull request by ctomc**

### CODE REVIEW

File: batch/extension-jberet/src/main/java/org/wildfly/extension/batch/jberet/BatchSubsystemDefinition.java [hunk 1]
```diff
@@ -84,7 +84,7 @@
     static final SimpleAttributeDefinition DEFAULT_THREAD_POOL = SimpleAttributeDefinitionBuilder.create("default-thread-pool", ModelType.STRING, true)
             .setAllowExpression(false)
             .setAttributeGroup("environment")
-            .setAttributeMarshaller(NameAttributeMarshaller.INSTANCE)
+            .setAttributeMarshaller(NameAttributeMarshaller.INSTANCE) //todo this looks like it is missing capability reference
```
Comment (jamezp) [inline batch/extension-jberet/src/main/java/org/wildfly/extension/batch/jberet/BatchSubsystemDefinition.java]: How can a capability be added for this? Unless something changed with the thread pool utilities in core you can't register them as capabilities.

Comment (ctomc) [inline batch/extension-jberet/src/main/java/org/wildfly/extension/batch/jberet/BatchSubsystemDefinition.java]: Add handler adds two capabilities and one of them looks like should have reference defined here
```
 private BatchSubsystemAdd() {
            super(Stream.of(Capabilities.DEFAULT_JOB_REPOSITORY_CAPABILITY, Capabilities.DEFAULT_THREAD_POOL_CAPABILITY).collect(Collectors.toSet()), 
DEFAULT_JOB_REPOSITORY, DEFAULT_THREAD_POOL);
        }
```
attribute DEFAULT_JOB_REPOSITORY has reference, but this one doesn't.

Comment (jamezp) [inline batch/extension-jberet/src/main/java/org/wildfly/extension/batch/jberet/BatchSubsystemDefinition.java]: The JavaDoc says:
> Records that this attribute's value represents a reference to an instance of a dynamic capability.

The `thread-pool` resource does not have a capability so I don't see how one could be referenced here. So while the `DEFAULT_THREAD_POOL` is a capability it doesn't not reference another capability.

Comment (ctomc) [inline batch/extension-jberet/src/main/java/org/wildfly/extension/batch/jberet/BatchSubsystemDefinition.java]: But it does reference a service itself.
Proper solution would probably be to add support for capabilities to Threads subsystem.

Comment (jamezp) [inline batch/extension-jberet/src/main/java/org/wildfly/extension/batch/jberet/BatchSubsystemDefinition.java]: Right. Service != capability though :)

Yes, I started that, but just didn't have time. It wouldn't be all that difficult, just time consuming. Really the main comment is I'd prefer not add a TODO for something that may or may not be fixable due to the how the thread pool resources work.

Comment (ctomc) [inline batch/extension-jberet/src/main/java/org/wildfly/extension/batch/jberet/BatchSubsystemDefinition.java]: removed the todo & rebased

**Comment by ctomc** [comment]
unrelated errors, but still.
retest this please

File: clustering/jgroups/extension/src/main/java/org/jboss/as/clustering/jgroups/subsystem/ProtocolResourceDefinition.java [hunk 2]
```diff
@@ -278,6 +278,12 @@ public void registerAttributes(ManagementResourceRegistration registration) {
         }
     }
 
+    @Override
+    public void registerCapabilities(ManagementResourceRegistration resourceRegistration) {
+        super.registerCapabilities(resourceRegistration);
+        resourceRegistration.registerCapability(Capability.SOCKET_BINDING.definition);
+    }
+
```
Comment (pferraro) [inline clustering/jgroups/extension/src/main/java/org/jboss/as/clustering/jgroups/subsystem/ProtocolResourceDefinition.java]: This capability is already registered with the resource. See:
https://github.com/wildfly/wildfly/blob/master/clustering/common/src/main/java/org/jboss/as/clustering/controller/AddStepHandler.java#L103

File: clustering/jgroups/extension/src/main/java/org/jboss/as/clustering/jgroups/subsystem/TransportResourceDefinition.java [hunk 3]
```diff
@@ -247,6 +247,12 @@ public void registerChildren(ManagementResourceRegistration registration) {
     }
 
     @Override
+    public void registerCapabilities(ManagementResourceRegistration resourceRegistration) {
+        super.registerCapabilities(resourceRegistration);
+        resourceRegistration.registerCapability(Capability.DIAGNOSTICS_SOCKET_BINDING.definition);
+    }
+
+    @Override
```
Comment (pferraro) [inline clustering/jgroups/extension/src/main/java/org/jboss/as/clustering/jgroups/subsystem/TransportResourceDefinition.java]: This capability is already registered with the resource.  See above.

File: clustering/singleton/extension/src/main/java/org/wildfly/extension/clustering/singleton/ElectionPolicyResourceDefinition.java [hunk 4]
```diff
@@ -109,6 +109,12 @@ public void registerAttributes(ManagementResourceRegistration registration) {
     }
 
     @Override
+    public void registerCapabilities(ManagementResourceRegistration resourceRegistration) {
+        super.registerCapabilities(resourceRegistration);
+        resourceRegistration.registerCapability(Capability.SOCKET_BINDING_PREFERENCE.definition);
+    }
+
+    @Override
```
Comment (pferraro) [inline clustering/singleton/extension/src/main/java/org/wildfly/extension/clustering/singleton/ElectionPolicyResourceDefinition.java]: This capability is already registered with the resource.  See above.

File: clustering/singleton/extension/src/main/java/org/wildfly/extension/clustering/singleton/SingletonPolicyResourceDefinition.java [hunk 5]
```diff
@@ -107,6 +107,12 @@ public void registerOperations(ManagementResourceRegistration registration) {
     }
 
     @Override
+    public void registerCapabilities(ManagementResourceRegistration resourceRegistration) {
+        super.registerCapabilities(resourceRegistration);
+        resourceRegistration.registerCapability(Capability.POLICY.definition);
+    }
+
+    @Override
```
Comment (pferraro) [inline clustering/singleton/extension/src/main/java/org/wildfly/extension/clustering/singleton/SingletonPolicyResourceDefinition.java]: This capability is already registered with the resource.  See above.

**Pull request merged**